src/harvester/doris_bfs.rs

```diff
@@ -146,12 +146,17 @@ async fn fetch_dataset(dir: &Dir, client: &Client, source: &Source, handle: &str
 
     let dataset = Dataset {
         title,
-        description: r#abstract,
+        description: Some(r#abstract),
```

Comment:
I think you just want to remove the `unwrap_or_default()` call in line 143 above as this is already optional in the source document.

Reply:
I tried but it did not compile - I got an Option<&str> but I want Option<String>. How do I get this?

Reply:
You still need keep the `.to_owned()`, but probably wrapped inside a `.map`, e.g.

```rust
.map(ToOwned::to_owned)
```

or if that does not work the slightly more complex

```rust
.map(|val| val.to_owned())
```